src/widgets/rectangle.rs

```diff
@@ -2,17 +2,23 @@
 // See the COPYRIGHT file at the top-level directory of this distribution.
 // Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
 
-use ffi::PangoRectangle;
+use std::mem;
+use glib::translate::*;
+use ffi;
 use libc::c_int;
-//use std::default::Default;
 
-pub trait Rectangle {
-    fn new(x: i32, y: i32, width: i32, height: i32) -> Self;
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+#[repr(C)]
+pub struct Rectangle {
+    pub x: i32,
+    pub y: i32,
+    pub width: i32,
+    pub height: i32,
 }
 
-impl Rectangle for PangoRectangle {
-    fn new(x: i32, y: i32, width: i32, height: i32) -> PangoRectangle {
-        PangoRectangle {
+impl Rectangle {
+    pub fn new(x: i32, y: i32, width: i32, height: i32) -> Rectangle {
+        Rectangle {
             x: x as c_int,
```

Comment:
Old code created PangoRectangle with c_int fields, you add struct with i32 fields